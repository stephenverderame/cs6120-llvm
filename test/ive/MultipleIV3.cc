// NOLINTBEGIN
#include <cstdio>
int main()
{
    int m = 20;
    int k = 100;
    int p = 0;
    for (int i = 5000; i >= m; i -= m) {
        printf("%d\n", k);
        k += m;
        int j = k * m + m;
        p = k + 10;
        int n = k * 10;
        int nn = n + m;
        printf("%d\n", j);
        printf("%d\n", i);
        printf("%d\n", p);
        printf("%d\n", n);
        printf("%d\n", nn);
    }
}
// NOLINTEND