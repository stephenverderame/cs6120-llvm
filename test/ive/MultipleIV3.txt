100
2420
5000
130
1200
1220
120
2820
4980
150
1400
1420
140
3220
4960
170
1600
1620
160
3620
4940
190
1800
1820
180
4020
4920
210
2000
2020
200
4420
4900
230
2200
2220
220
4820
4880
250
2400
2420
240
5220
4860
270
2600
2620
260
5620
4840
290
2800
2820
280
6020
4820
310
3000
3020
300
6420
4800
330
3200
3220
320
6820
4780
350
3400
3420
340
7220
4760
370
3600
3620
360
7620
4740
390
3800
3820
380
8020
4720
410
4000
4020
400
8420
4700
430
4200
4220
420
8820
4680
450
4400
4420
440
9220
4660
470
4600
4620
460
9620
4640
490
4800
4820
480
10020
4620
510
5000
5020
500
10420
4600
530
5200
5220
520
10820
4580
550
5400
5420
540
11220
4560
570
5600
5620
560
11620
4540
590
5800
5820
580
12020
4520
610
6000
6020
600
12420
4500
630
6200
6220
620
12820
4480
650
6400
6420
640
13220
4460
670
6600
6620
660
13620
4440
690
6800
6820
680
14020
4420
710
7000
7020
700
14420
4400
730
7200
7220
720
14820
4380
750
7400
7420
740
15220
4360
770
7600
7620
760
15620
4340
790
7800
7820
780
16020
4320
810
8000
8020
800
16420
4300
830
8200
8220
820
16820
4280
850
8400
8420
840
17220
4260
870
8600
8620
860
17620
4240
890
8800
8820
880
18020
4220
910
9000
9020
900
18420
4200
930
9200
9220
920
18820
4180
950
9400
9420
940
19220
4160
970
9600
9620
960
19620
4140
990
9800
9820
980
20020
4120
1010
10000
10020
1000
20420
4100
1030
10200
10220
1020
20820
4080
1050
10400
10420
1040
21220
4060
1070
10600
10620
1060
21620
4040
1090
10800
10820
1080
22020
4020
1110
11000
11020
1100
22420
4000
1130
11200
11220
1120
22820
3980
1150
11400
11420
1140
23220
3960
1170
11600
11620
1160
23620
3940
1190
11800
11820
1180
24020
3920
1210
12000
12020
1200
24420
3900
1230
12200
12220
1220
24820
3880
1250
12400
12420
1240
25220
3860
1270
12600
12620
1260
25620
3840
1290
12800
12820
1280
26020
3820
1310
13000
13020
1300
26420
3800
1330
13200
13220
1320
26820
3780
1350
13400
13420
1340
27220
3760
1370
13600
13620
1360
27620
3740
1390
13800
13820
1380
28020
3720
1410
14000
14020
1400
28420
3700
1430
14200
14220
1420
28820
3680
1450
14400
14420
1440
29220
3660
1470
14600
14620
1460
29620
3640
1490
14800
14820
1480
30020
3620
1510
15000
15020
1500
30420
3600
1530
15200
15220
1520
30820
3580
1550
15400
15420
1540
31220
3560
1570
15600
15620
1560
31620
3540
1590
15800
15820
1580
32020
3520
1610
16000
16020
1600
32420
3500
1630
16200
16220
1620
32820
3480
1650
16400
16420
1640
33220
3460
1670
16600
16620
1660
33620
3440
1690
16800
16820
1680
34020
3420
1710
17000
17020
1700
34420
3400
1730
17200
17220
1720
34820
3380
1750
17400
17420
1740
35220
3360
1770
17600
17620
1760
35620
3340
1790
17800
17820
1780
36020
3320
1810
18000
18020
1800
36420
3300
1830
18200
18220
1820
36820
3280
1850
18400
18420
1840
37220
3260
1870
18600
18620
1860
37620
3240
1890
18800
18820
1880
38020
3220
1910
19000
19020
1900
38420
3200
1930
19200
19220
1920
38820
3180
1950
19400
19420
1940
39220
3160
1970
19600
19620
1960
39620
3140
1990
19800
19820
1980
40020
3120
2010
20000
20020
2000
40420
3100
2030
20200
20220
2020
40820
3080
2050
20400
20420
2040
41220
3060
2070
20600
20620
2060
41620
3040
2090
20800
20820
2080
42020
3020
2110
21000
21020
2100
42420
3000
2130
21200
21220
2120
42820
2980
2150
21400
21420
2140
43220
2960
2170
21600
21620
2160
43620
2940
2190
21800
21820
2180
44020
2920
2210
22000
22020
2200
44420
2900
2230
22200
22220
2220
44820
2880
2250
22400
22420
2240
45220
2860
2270
22600
22620
2260
45620
2840
2290
22800
22820
2280
46020
2820
2310
23000
23020
2300
46420
2800
2330
23200
23220
2320
46820
2780
2350
23400
23420
2340
47220
2760
2370
23600
23620
2360
47620
2740
2390
23800
23820
2380
48020
2720
2410
24000
24020
2400
48420
2700
2430
24200
24220
2420
48820
2680
2450
24400
24420
2440
49220
2660
2470
24600
24620
2460
49620
2640
2490
24800
24820
2480
50020
2620
2510
25000
25020
2500
50420
2600
2530
25200
25220
2520
50820
2580
2550
25400
25420
2540
51220
2560
2570
25600
25620
2560
51620
2540
2590
25800
25820
2580
52020
2520
2610
26000
26020
2600
52420
2500
2630
26200
26220
2620
52820
2480
2650
26400
26420
2640
53220
2460
2670
26600
26620
2660
53620
2440
2690
26800
26820
2680
54020
2420
2710
27000
27020
2700
54420
2400
2730
27200
27220
2720
54820
2380
2750
27400
27420
2740
55220
2360
2770
27600
27620
2760
55620
2340
2790
27800
27820
2780
56020
2320
2810
28000
28020
2800
56420
2300
2830
28200
28220
2820
56820
2280
2850
28400
28420
2840
57220
2260
2870
28600
28620
2860
57620
2240
2890
28800
28820
2880
58020
2220
2910
29000
29020
2900
58420
2200
2930
29200
29220
2920
58820
2180
2950
29400
29420
2940
59220
2160
2970
29600
29620
2960
59620
2140
2990
29800
29820
2980
60020
2120
3010
30000
30020
3000
60420
2100
3030
30200
30220
3020
60820
2080
3050
30400
30420
3040
61220
2060
3070
30600
30620
3060
61620
2040
3090
30800
30820
3080
62020
2020
3110
31000
31020
3100
62420
2000
3130
31200
31220
3120
62820
1980
3150
31400
31420
3140
63220
1960
3170
31600
31620
3160
63620
1940
3190
31800
31820
3180
64020
1920
3210
32000
32020
3200
64420
1900
3230
32200
32220
3220
64820
1880
3250
32400
32420
3240
65220
1860
3270
32600
32620
3260
65620
1840
3290
32800
32820
3280
66020
1820
3310
33000
33020
3300
66420
1800
3330
33200
33220
3320
66820
1780
3350
33400
33420
3340
67220
1760
3370
33600
33620
3360
67620
1740
3390
33800
33820
3380
68020
1720
3410
34000
34020
3400
68420
1700
3430
34200
34220
3420
68820
1680
3450
34400
34420
3440
69220
1660
3470
34600
34620
3460
69620
1640
3490
34800
34820
3480
70020
1620
3510
35000
35020
3500
70420
1600
3530
35200
35220
3520
70820
1580
3550
35400
35420
3540
71220
1560
3570
35600
35620
3560
71620
1540
3590
35800
35820
3580
72020
1520
3610
36000
36020
3600
72420
1500
3630
36200
36220
3620
72820
1480
3650
36400
36420
3640
73220
1460
3670
36600
36620
3660
73620
1440
3690
36800
36820
3680
74020
1420
3710
37000
37020
3700
74420
1400
3730
37200
37220
3720
74820
1380
3750
37400
37420
3740
75220
1360
3770
37600
37620
3760
75620
1340
3790
37800
37820
3780
76020
1320
3810
38000
38020
3800
76420
1300
3830
38200
38220
3820
76820
1280
3850
38400
38420
3840
77220
1260
3870
38600
38620
3860
77620
1240
3890
38800
38820
3880
78020
1220
3910
39000
39020
3900
78420
1200
3930
39200
39220
3920
78820
1180
3950
39400
39420
3940
79220
1160
3970
39600
39620
3960
79620
1140
3990
39800
39820
3980
80020
1120
4010
40000
40020
4000
80420
1100
4030
40200
40220
4020
80820
1080
4050
40400
40420
4040
81220
1060
4070
40600
40620
4060
81620
1040
4090
40800
40820
4080
82020
1020
4110
41000
41020
4100
82420
1000
4130
41200
41220
4120
82820
980
4150
41400
41420
4140
83220
960
4170
41600
41620
4160
83620
940
4190
41800
41820
4180
84020
920
4210
42000
42020
4200
84420
900
4230
42200
42220
4220
84820
880
4250
42400
42420
4240
85220
860
4270
42600
42620
4260
85620
840
4290
42800
42820
4280
86020
820
4310
43000
43020
4300
86420
800
4330
43200
43220
4320
86820
780
4350
43400
43420
4340
87220
760
4370
43600
43620
4360
87620
740
4390
43800
43820
4380
88020
720
4410
44000
44020
4400
88420
700
4430
44200
44220
4420
88820
680
4450
44400
44420
4440
89220
660
4470
44600
44620
4460
89620
640
4490
44800
44820
4480
90020
620
4510
45000
45020
4500
90420
600
4530
45200
45220
4520
90820
580
4550
45400
45420
4540
91220
560
4570
45600
45620
4560
91620
540
4590
45800
45820
4580
92020
520
4610
46000
46020
4600
92420
500
4630
46200
46220
4620
92820
480
4650
46400
46420
4640
93220
460
4670
46600
46620
4660
93620
440
4690
46800
46820
4680
94020
420
4710
47000
47020
4700
94420
400
4730
47200
47220
4720
94820
380
4750
47400
47420
4740
95220
360
4770
47600
47620
4760
95620
340
4790
47800
47820
4780
96020
320
4810
48000
48020
4800
96420
300
4830
48200
48220
4820
96820
280
4850
48400
48420
4840
97220
260
4870
48600
48620
4860
97620
240
4890
48800
48820
4880
98020
220
4910
49000
49020
4900
98420
200
4930
49200
49220
4920
98820
180
4950
49400
49420
4940
99220
160
4970
49600
49620
4960
99620
140
4990
49800
49820
4980
100020
120
5010
50000
50020
5000
100420
100
5030
50200
50220
5020
100820
80
5050
50400
50420
5040
101220
60
5070
50600
50620
5060
101620
40
5090
50800
50820
5080
102020
20
5110
51000
51020
