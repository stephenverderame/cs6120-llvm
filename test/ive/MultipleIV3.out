Loop: 
1:                                                ; preds = %17, %0
  %.01 = phi i32 [ 100, %0 ], [ %6, %17 ]
  %.0 = phi i32 [ 5000, %0 ], [ %18, %17 ]
  %2 = icmp sge i32 %.0, 20
  br i1 %2, label %4, label %3

Derived IV: %10 = %6 * 10
Derived IV: %11 = %6 * 10 + 20
Basic IV:   %18 = sub nsw i32 %.0, 20
Basic IV:   %6 = add nsw i32 %.01, 20
Derived IV: %7 = %6 * 20
Derived IV: %8 = %6 * 20 + 20
Basic IV:   %.0 = phi i32 [ 5000, %0 ], [ %18, %17 ]
Basic IV:   %.01 = phi i32 [ 100, %0 ], [ %6, %17 ]
